File: backend/src/main/java/me/kzv/okvue/modules/reply/Reply.java
package me.kzv.okvue.modules.reply;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import me.kzv.okvue.modules.account.Account;
import me.kzv.okvue.modules.post.Post;

import javax.persistence.*;

@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Entity
public class Reply {

    @Id @GeneratedValue
    @Column(name = "reply_id")
    private Long id;

    private String text;

    @ManyToOne(fetch = FetchType.LAZY)
    private Account commenter;

    @ManyToOne(fetch = FetchType.LAZY)
    private Post post;

    @Builder
    public Reply(Long id, String text, Account commenter, Post post) {
        this.id = id;
        this.text = text;
        this.commenter = commenter;
        this.post = post;
    }
}

File: rest/backend/src/main/kotlin/me/kzv/board/web/domain/Reply.kt
package me.kzv.board.web.domain

import org.hibernate.annotations.GenericGenerator
import javax.persistence.*

@Entity
class Reply(
    @Id
    @GeneratedValue(generator = "sys-uuid")
    @GenericGenerator(name = "sys-uuid", strategy = "uuid")
    @Column(name = "reply_id")
    var id: String? = null,
    @Lob
    var text: String,
    var replyer: String,
    var replyType: Boolean, // false: 댓글 true: 대댓글
    var replyGroup: String? = null, // 대댓글일 경우 그룹 아이디가 있어야함

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    var post: Post,
) : BaseTimeEntity() {
}